src/main/scala/com/crealytics/spark/excel/ExcelRelation.scala

```diff
@@ -1,138 +1,161 @@
+    val result = rows.to[Vector]
+
+    sqlContext.sparkContext.parallelize(result.map(Row.fromSeq))
+  }
+
+  // Cast a String to a Spark Data Type
+  private def castTo(datum: String, castType: DataType): Any = {
+    if (datum == null) {
+      return null
+    }
+    castType match {
+      case _: ByteType => datum.toByte
+      case _: ShortType => datum.toShort
+      case _: IntegerType => datum.toInt
+      case _: LongType => datum.toLong
+      case _: FloatType => Try(datum.toFloat)
+        .getOrElse(NumberFormat.getInstance(Locale.getDefault).parse(datum).floatValue())
+      case _: DoubleType => Try(datum.toDouble)
+        .getOrElse(NumberFormat.getInstance(Locale.getDefault).parse(datum).doubleValue())
+      case _: BooleanType => datum.toBoolean
+      case _: DecimalType => new BigDecimal(datum.replaceAll(",", ""))
+      case _: TimestampType => Timestamp.valueOf(datum)
+      case _: DateType => Date.valueOf(datum)
+      case _: StringType => datum
+      case _ => throw new RuntimeException(s"Unsupported type: ${castType.typeName}")
+    }
+  }
+
+  private def rowsRdd: RDD[SheetRow] = {
+    parallelize(sheet.rowIterator().asScala.toSeq)
+  }
+
+  private def dataRows = sheet.rowIterator.asScala.drop(if (useHeader) 1 else 0)
+  private def parallelize[T: scala.reflect.ClassTag](seq: Seq[T]): RDD[T] = sqlContext.sparkContext.parallelize(seq)
+  private def inferSchema(): StructType = {
+    if (this.userSchema != null) {
+      userSchema
+    } else {
+      val firstRow = headers
+      val header = if (useHeader) {
+        firstRow.map(_.getStringCellValue)
+      } else {
+        firstRow.zipWithIndex.map { case (value, index) => s"C$index"}
+      }
+      val baseSchema = if (this.inferSheetSchema) {
+        val stringsAndCellTypes = dataRows.map(_.cellIterator.asScala.map(c => c.getCellType).toVector).toVector
+        InferSchema(parallelize(stringsAndCellTypes), header.toArray)
+      } else {
+        // By default fields are assumed to be StringType
+        val schemaFields = header.map { fieldName =>
+          StructField(fieldName.toString, StringType, nullable = true)
+        }
+        StructType(schemaFields)
+      }
+      if (addColorColumns) {
+        header.foldLeft(baseSchema) {(schema, header) => schema.add(s"${header}_color", StringType, nullable = true)}
+      } else {
+        baseSchema
+      }
+    }
+  }
+  
+  /**
+   * Returns the header of a sheet skipping initial empty rows.
+   * 
+   * @param The sheet for which the header should be returned
+   * @return Vector[Cell]
+   */
+  private def getHeaders(sheet: Sheet): Vector[Cell] = {
+    var headers: Vector[Cell] = Vector.empty
+    
+    var rowIndex = 0
+    var isRowEmpty = true
+    while(isRowEmpty) {
+      if(sheet.getRow(rowIndex) != null) {
+        isRowEmpty = false
+        headers = sheet.getRow(rowIndex).cellIterator().asScala.to[Vector]
+      }
+      rowIndex += 1
+    }
+    
+    return headers
+  }
+}

```

Comment:
It seems that there's a lot of whitespace change here. Do you happen to be on Windows and have not set `autocrlf`?
```
git config --global core.autocrlf true
```